controllers/databaseclusterbackup_controller.go

```diff
@@ -153,25 +160,206 @@ func (r *DatabaseClusterBackupReconciler) SetupWithManager(mgr ctrl.Manager) err
 	if err == nil {
 		if err := r.addPXCToScheme(r.Scheme); err == nil {
 			controller.Owns(&pxcv1.PerconaXtraDBClusterBackup{})
+			controller.Watches(
+				&pxcv1.PerconaXtraDBClusterBackup{},
+				handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+					return r.tryCreateDBBackups(ctx, obj, r.tryCreatePXC)
+				}))
 		}
 	}
 
 	err = r.Get(ctx, types.NamespacedName{Name: psmdbBackupCRDName}, unstructuredResource)
 	if err == nil {
 		if err := r.addPSMDBToScheme(r.Scheme); err == nil {
 			controller.Owns(&psmdbv1.PerconaServerMongoDBBackup{})
+			controller.Watches(
+				&psmdbv1.PerconaServerMongoDBBackup{},
+				handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+					return r.tryCreateDBBackups(ctx, obj, r.tryCreatePSMDB)
+				}))
 		}
 	}
 
 	err = r.Get(ctx, types.NamespacedName{Name: pgBackupCRDName}, unstructuredResource)
 	if err == nil {
 		if err := r.addPGToScheme(r.Scheme); err == nil {
 			controller.Owns(&pgv2.PerconaPGBackup{})
+			controller.Watches(
+				&pgv2.PerconaPGBackup{},
+				handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+					return r.tryCreateDBBackups(ctx, obj, r.tryCreatePG)
+				}))
 		}
 	}
 	return controller.Complete(r)
 }
 
+func (r *DatabaseClusterBackupReconciler) tryCreateDBBackups(
+	ctx context.Context,
+	obj client.Object,
+	createBackupFunc func(ctx context.Context, obj client.Object) error,
+) []reconcile.Request {
+	logger := log.FromContext(ctx)
+	if len(obj.GetOwnerReferences()) == 0 {
+		err := createBackupFunc(ctx, obj)
+		if err != nil {
+			logger.Error(err, fmt.Sprintf("Failed to create DatabaseClusterBackup %s", obj.GetName()))
+		}
+	}
+
+	return []reconcile.Request{{
+		NamespacedName: types.NamespacedName{
+			Name:      obj.GetName(),
+			Namespace: obj.GetNamespace(),
+		},
+	}}
+}
+
+func (r *DatabaseClusterBackupReconciler) tryCreatePG(ctx context.Context, obj client.Object) error {
+	pgBackup := &pgv2.PerconaPGBackup{}
+	namespacedName := types.NamespacedName{
+		Namespace: obj.GetNamespace(),
+		Name:      obj.GetName(),
+	}
+
+	if err := r.Get(ctx, namespacedName, pgBackup); err != nil {
+		// if such upstream backup is not found - do nothing
+		if k8serrors.IsNotFound(err) {
+			return nil
```

Comment:
Do we need to check if we shall remove our `DatabaseClusterBackup` in this case?

Reply:
This is a good point. We actually have a common scenario that illustrates this nicely. We set the retention copies, i.e. number of backups that are kept, and when a new backup exceeds that number the oldest backup will be removed and in that case we need to remove the corresponding DBBackup.
We must have a 1-1 match between DBBackup and the operator's backup CRs.

Reply:
This is a really good point that have prompted some thoughts and discussions. 

We have this `RetentionCopies` setting which means that at some point the upstream backup object could be deleted by the upstream operator. In this case we need to delete the everest backup CR too. 

I explored the option to set a circular dependency: everest CR owns the upsream CR and vice versa, so when one is deleted, the other one gets deleted automatically. It's not forbidden by any documentation and as it turned out it works well. 
So I added the backward "upstream CR -> everest CR" ownership relation too so we don't need to worry about the deletion.